platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,993 @@
+   * key of the `tssEncryptionKey` of the node sending this transaction.
+   */
+  bytes signature = 6;
+}
+```
+
+##### TssVote System Transaction
+
+```protobuf
+/**
+ * The TssVoteTransaction is used to vote on the validity of TssMessages for a candidate roster.
+ */
+message TssVoteTransaction {
+
+  /**
+   * The node id is the id of the node that generated the TssVote.
+   */
+  uint64 node_id = 1;
+
+  /**
+   * The source roster hash is the hash of the roster containing the node generating the TssVote.
+   */
+  bytes source_roster_hash = 2;
+
+  /**
+   * The target roster hash is the hash of the roster that the TssVote is for.
+   */
+  bytes target_roster_hash = 3;
+
+  /**
+   * The ledger id that was computed from the TssMessages for the target roster.
+   */
+  bytes ledger_id = 4;
+
+  /**
+   * The signature produced by the node using its RSA signing key to sign the ledger_id.  This signature is used to
+   * establish a chain of trust in the ledger id.
+   */
+  bytes node_signature = 5;
+
+  /**
+   * The tss_vote is a bit vector where the index position of each bit corresponds to the sequence number of a
+   * `TssMessageTransaction`. The least significant bit of byte[0] corresponds to sequence number 0 and the most
+   * significant bit of byte[0] corresponds to sequence number 7.  The least significant bit of byte[1] corresponds to
+   * sequence number 8 and the most significant bit of byte[1] corresponds to the sequence number 15, and so on.
+   * If a bit is set to 1, then the TssMessage for the corresponding TssMessageTransaction sequence number was
+   * received as one of the first Threshold number of valid TssMessages in consensus order.  If a bit is set to 0,
+   * then the TssMessage was invalid or there was no TssMessageTransaction with a sequence number corresponding to
+   * the bit index.
+   */
+  bytes tss_vote = 5;
+
+  /**
+   * The signature is produced by the node signing the tuple (source_roster_hash, target_roster_hash, ledger_id,
+   * node_signature, tss_vote) using its private `tssEncryptionKey`.  The signature is verified by other nodes using
+   * the public key of the `tssEncryptionKey` of the node sending this transaction.
+   */
+  bytes signature = 6;
+}
+```
+
+#### State Datastructures
+
+The state needs to store the relevant ledger id and TSS Data (key material and votes).
+
+##### Ledger Id Queue
+
+The `ledgerId` is the public ledger key able to verify ledger signatures. It is used by the platform and other
+entities outside the consensus node. This value should not change during normal address book changes and its value does
+not change unless the network goes through another TSS bootstrap process. When the value is changes, to transfer trust,
+the old ledger private key should sign the new ledger id. This signature should be stored along with the round
+value for when the new ledger id becomes active. Triples of ledger id, signature, and round are stored in a queue in
```

Comment:
I'm changing it to list and updating the description.

Reply:
Each time the ledger id changes is a critical event.  Ideally it will happen only a couple times and due to underlying rotation of technology.   I get your argument, but given it is low quantity of data that rarely changes and is foundational to our ability to trust, it makes sense to me to model it in the state.   But I will ask for specific review from Leemon and Jasper

Reply:
the size of the collection is potentially infinite.  But practically it shouldn't reach a size of 10 in my life time.  I would hope it doesn't reach a size of 5 during my employment. :-).   Maybe 3. 
1 - initial assignment
2 - possible hinTS like replacement
3 - post quantum solution.

Reply:
Makes sense. Thanks for the additional info.